File: scripts/migrations/run-investment-migration.js
/**
 * Run Investment Tables Migration
 * This script executes the SQL migration directly on Supabase
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const SUPABASE_URL = 'https://diexsbzqwsbpilsymnfb.supabase.co';
const SUPABASE_SERVICE_KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImRpZXhzYnpxd3NicGlsc3ltbmZiIiwicm9sZSI6InNlcnZpY2Vfcm9sZSIsImlhdCI6MTc2MDM5MjE5MSwiZXhwIjoyMDc1OTY4MTkxfQ.30ZRAfvIyQUBzyf3xqvrwXbeR15FXDnTGVvTfwmeEXY';

console.log('🚀 Investment Tables Migration');
console.log('='.repeat(60));
console.log('');

// Read migration file
const migrationPath = path.join(__dirname, 'supabase', 'migrations', '20241113_investment_tables.sql');

console.log('📄 Reading migration file...');
let sql;
try {
  sql = fs.readFileSync(migrationPath, 'utf8');
  console.log(`✅ Loaded ${sql.length} characters`);
} catch (error) {
  console.error('❌ Error reading migration file:', error.message);
  process.exit(1);
}

console.log('');
console.log('📡 Executing SQL on Supabase...');

// Execute SQL via Supabase REST API
const executeSQL = async () => {
  try {
    const response = await fetch(`${SUPABASE_URL}/rest/v1/rpc/exec_sql`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'apikey': SUPABASE_SERVICE_KEY,
        'Authorization': `Bearer ${SUPABASE_SERVICE_KEY}`,
        'Prefer': 'return=representation'
      },
      body: JSON.stringify({
        query: sql
      })
    });

    if (!response.ok) {
      const errorText = await response.text();
      throw new Error(`HTTP ${response.status}: ${errorText}`);
    }

    const result = await response.json();
    console.log('✅ Migration executed successfully!');
    console.log('');
    console.log('📊 Result:', JSON.stringify(result, null, 2));
    
  } catch (error) {
    console.error('❌ Error executing migration:', error.message);
    console.log('');
    console.log('💡 Alternative: Use Supabase SQL Editor');
    console.log('   1. Go to: https://supabase.com/dashboard/project/diexsbzqwsbpilsymnfb/sql/new');
    console.log('   2. Copy SQL from: supabase/migrations/20241113_investment_tables.sql');
    console.log('   3. Click "Run"');
    process.exit(1);
  }
};

executeSQL().then(() => {
  console.log('');
  console.log('🎉 Migration complete!');
  console.log('');
  console.log('📋 Next Steps:');
  console.log('   1. Test form: http://localhost:8080/project-showcase/sabo-arena/investment/apply');
  console.log('   2. Check data: https://supabase.com/dashboard/project/diexsbzqwsbpilsymnfb/editor');
  console.log('   3. API endpoint: http://localhost:3001/api/investment/applications');
});

File: admin/scripts/migrations/run_investment_migration.py
#!/usr/bin/env python3
"""
Execute Investment Tables Migration on Supabase
"""

import psycopg2
import os
from pathlib import Path

# Supabase connection details - Direct connection (no pooler)
DB_HOST = "diexsbzqwsbpilsymnfb.supabase.co"
DB_PORT = 5432
DB_NAME = "postgres"
DB_USER = "postgres"
DB_PASSWORD = "Acookingoil123"

print("🚀 Investment Tables Migration")
print("=" * 70)
print()

# Read migration SQL
migration_file = Path(__file__).parent / "supabase" / "migrations" / "20241113_investment_tables.sql"

print(f"📄 Reading migration file: {migration_file}")
try:
    with open(migration_file, 'r', encoding='utf-8') as f:
        sql = f.read()
    print(f"✅ Loaded {len(sql)} characters")
except Exception as e:
    print(f"❌ Error reading file: {e}")
    exit(1)

print()
print("📡 Connecting to Supabase...")

# Connect to database
connection_string = f"postgresql://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{DB_PORT}/{DB_NAME}?sslmode=require"

try:
    conn = psycopg2.connect(connection_string)
    print("✅ Connected successfully!")
except Exception as e:
    print(f"❌ Connection failed: {e}")
    exit(1)

print()
print("⚙️  Executing SQL migration...")
print("-" * 70)

# Execute migration
cursor = conn.cursor()
try:
    cursor.execute(sql)
    conn.commit()
    print()
    print("✅ Migration executed successfully!")
    
    # Verify tables
    print()
    print("🔍 Verifying tables...")
    
    cursor.execute("""
        SELECT table_name 
        FROM information_schema.tables 
        WHERE table_schema = 'public' 
        AND table_name IN ('investment_applications', 'project_interests')
        ORDER BY table_name;
    """)
    
    tables = cursor.fetchall()
    print()
    for table in tables:
        print(f"   ✅ {table[0]}")
    
    # Count records
    print()
    print("📊 Table Statistics:")
    
    cursor.execute("SELECT COUNT(*) FROM investment_applications")
    app_count = cursor.fetchone()[0]
    print(f"   • investment_applications: {app_count} records")
    
    cursor.execute("SELECT COUNT(*) FROM project_interests")
    int_count = cursor.fetchone()[0]
    print(f"   • project_interests: {int_count} records")
    
    print()
    print("🎉 Migration completed successfully!")
    print()
    print("📋 Next Steps:")
    print("   1. Test form: http://localhost:8080/project-showcase/sabo-arena/investment/apply")
    print("   2. Check data: https://supabase.com/dashboard/project/diexsbzqwsbpilsymnfb/editor")
    print("   3. API endpoint: http://localhost:3001/api/investment/applications")
    
except psycopg2.Error as e:
    conn.rollback()
    print()
    print(f"❌ Error executing migration: {e}")
    print()
    print("💡 Tip: Check if tables already exist or if there are permission issues")
    exit(1)
    
except Exception as e:
    conn.rollback()
    print()
    print(f"❌ Unexpected error: {e}")
    exit(1)
    
finally:
    cursor.close()
    conn.close()
    print()
    print("🔌 Connection closed")
